src/adapter-loader.js

```diff
@@ -11,10 +11,74 @@
 'use strict';
 
 var config = require('config');
-const Constants = require('./constants');
 const GetOpt = require('node-getopt');
 const PluginClient = require('./adapters/plugin/plugin-client');
+const fs = require('fs');
+const path = require('path');
 
+function loadAdapter(adapterId, verbose) {
+  const adapterPath = path.join(__dirname,
+                                config.get('adapterManager.path'),
+                                adapterId);
+
+  // Skip if there's no package.json file.
+  const packageJson = path.join(adapterPath, 'package.json');
+  if (!fs.lstatSync(packageJson).isFile()) {
+    console.error('package.json not found for adapter:', adapterId);
+    return false;
+  }
+
+  // Read the package.json file.
+  let data;
+  try {
+    data = fs.readFileSync(packageJson);
+  } catch (err) {
+    console.error('Failed to read package.json for adapter:', adapterId);
+    console.error(err);
+    return false;
+  }
+
+  let manifest;
+  try {
+    manifest = JSON.parse(data);
+  } catch (err) {
+    console.error('Failed to parse package.json for adapter:', adapterId);
+    console.error(err);
+    return false;
+  }
+
+  // Verify API version.
+  const apiVersion = config.get('adapterManager.api');
+  if (manifest.moziot.api.min > apiVersion ||
+      manifest.moziot.api.max < apiVersion) {
+    console.error('API mismatch for adapter:', manifest.name);
+    console.error('Current:', apiVersion, 'Supported:',
+                  manifest.moziot.api.min, '-', manifest.moziot.api.max);
+    return false;
+  }
+  let newSettings = Object.assign({}, manifest);
+  if (!newSettings.moziot.hasOwnProperty('config')) {
+    newSettings.moziot.config = {};
+  }
+
+  const errorCallback = function(packageName, errorStr) {
+    console.log('Failed to load', packageName, '-', errorStr);
+  };
+
+  var pluginClient = new PluginClient(adapterId, {verbose: verbose});
+  pluginClient.register().then(adapterManagerProxy => {
```

Comment:
Once the code reaches this point it will always return true because it doesn't do anything with this promise (either returning or awaiting it)